third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1257,11 +1298,23 @@ void populateElementwiseOpToLLVMPatterns(
   patterns.add<FpToFpOpConversion>(typeConverter, axisInfoAnalysis,
                                    targetInfo.getISAFamily(), benefit);
 
-  // ExpOpConversionApprox will try using ex2.approx if the input type is
+  // ExpOpConversionApprox will try using __ocml_exp2_f32 if the input type is
   // FP32. For other input types, ExpOpConversionApprox will return failure and
   // ElementwiseOpConversion<math::ExpOp, math::ExpOp> defined below will call
-  // __nv_expf for higher-precision calculation
+  // __ocml_exp_f64 for higher-precision calculation
   patterns.add<ExpOpConversionApprox>(typeConverter, axisInfoAnalysis, benefit);
+  // Exp2OpConversion will use __ocml_exp2_f32 or __ocml_native_exp2_f32
+  // based on the __HIP_FTZ flag if the input type if FP32. For FP64 input,
+  // Exp2OpConversion will return failure and later pass will call
+  // __ocml_exp2_f64 for higher-precision calculation
+  // The denorm flushing behavior is as follows:
+  // 1. If __HIP_FTZ = 1, exp2 flushes denorms in input and output regardless
+  //    of the value of kernel arg `allow_flush_denorm`.
+  // 2. If __HIP_FTZ = 0, whether exp2 flushes denorms in input and output
+  //    depends on the value of kernel arg `allow_flush_denorm`.
+  // 3. __HIP_FTZ is default to 1 and not exposed as a kernel argument.
```

Comment:
At least for clang, this is subtarget dependent. Anything gfx9+ defaults to not flushing f32

Reply:
I set the default to flushing to match the behavior on the NV path.

Reply:
Did you check with different SM version targets? I think they also switch the default at some level

Reply:
From NV libdevice, they use `__CUDA_FTZ` to control whether `.ftz` should be used. I checked the ptx and sass on A100. I can check H100 to make sure this is also the case there.